tests/zfs-tests/tests/functional/zvol/zvol_misc/issue-6238.ksh

```diff
@@ -0,0 +1,69 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright 2017, loli10K <ezomori.nozomu@gmail.com>. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/cli_root/zfs_set/zfs_set_common.kshlib
+. $STF_SUITE/tests/functional/zvol/zvol_common.shlib
+. $STF_SUITE/tests/functional/zvol/zvol_misc/zvol_misc_common.kshlib
+
+#
+# DESCRIPTION:
+# Verify we can reproduce issue #6238 on the buildbot
+#
+# STRATEGY:
+# 1. Create a ZVOL with logbias=throughput and sync=always
+# 2. Write data to its device node
+# 3. Verify we hit "panic: allocating allocated segment"
+#
+
+verify_runnable "global"
+
+function cleanup
+{
+	# FIXME: remove after debugging
+	cat /proc/spl/taskq-all
+	datasetexists $ZVOL && log_must_busy zfs destroy -r $ZVOL
+	block_device_wait
+}
+
+log_assert "Verify we can reproduce issue #6238 on the buildbot"
+log_onexit cleanup
+
+ZVOL="$TESTPOOL/vol"
+ZDEV="$ZVOL_DEVDIR/$ZVOL"
+
+# 1. Create a ZVOL with logbias=throughput and sync=always
+log_must zfs create -V $VOLSIZE -b 128K -o sync=always -o logbias=throughput $ZVOL
+
+# 2. Write data to its device node
+for i in {1..50}; do
+	dd if=/dev/zero of=$ZDEV bs=8k count=1 &
+done
+
+# 3. Verify we hit "panic: allocating allocated segment"
```

Comment:
Ahh, I see.   Well since this is still a real issue it'd be nice to keep a test case for the fix.